src/zjs_callbacks.c

```diff
@@ -369,6 +369,7 @@ void zjs_remove_all_callbacks()
         CB_LOCK();
         if (cb_map[i]) {
             zjs_remove_callback_priv(i, skip_flush);
+            zjs_free_callback(i);
```

Comment:
No, this isn't right... if you look in zjs_remove_callback_prev it may free the callback itself under some circumstances. You don't want to do it twice.

If you need to try to force all cleanup synchronously and can't wait for things to flush out naturally, I think what you probably need to do, which came up in our discussion yesterday, is remove all timers / callbacks and then call zjs_service_callbacks, probably in your ashell cleanup routine I'm guessing? 

How's that for a sentence!

-1 for now

Reply:
I tried zjs_service_callbacks in several places, but it doesn't solve the issue.  Something in the ring buffer remains at the end. This was the only method I found that successfully got rid of all the references from the callbacks. Afraid I don't understand the ring buffer well, otherwise I'd tell you why its happening.

Reply:
I could change it so that zjs_remove_callback_priv is passed skip_flush = true, then we'd be sure it didn't get freed 2x